php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1402 @@
+        .is(f.tryStatement(
+            b.token(PHPKeyword.TRY),
+            BLOCK(),
+            b.zeroOrMore(CATCH_BLOCK()),
+            b.optional(f.newTuple2(b.token(PHPKeyword.FINALLY), BLOCK()))));
+  }
+
+  public CatchBlockTree CATCH_BLOCK() {
+    return b.<CatchBlockTree>nonterminal(PHPLexicalGrammar.CATCH_BLOCK)
+        .is(f.catchBlock(
+            b.token(PHPKeyword.CATCH),
+            b.token(PHPPunctuator.LPARENTHESIS),
+            NAMESPACE_NAME(),
+            b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+            b.token(PHPPunctuator.RPARENTHESIS),
+            BLOCK()
+        ));
+
+  }
+
+  public BlockTree BLOCK() {
+    return b.<BlockTree>nonterminal(PHPLexicalGrammar.BLOCK)
+        .is(f.block(
+            b.token(PHPPunctuator.LCURLYBRACE),
+            b.zeroOrMore(INNER_STATEMENT()),
+            b.token(PHPPunctuator.RCURLYBRACE)
+        ));
+  }
+
+  public GotoStatementTree GOTO_STATEMENT() {
+    return b.<GotoStatementTree>nonterminal(PHPLexicalGrammar.GOTO_STATEMENT)
+        .is(f.gotoStatement(b.token(PHPKeyword.GOTO), b.token(PHPLexicalGrammar.IDENTIFIER), EOS()));
+  }
+
+  public ExpressionStatementTree EXPRESSION_STATEMENT() {
+    return b.<ExpressionStatementTree>nonterminal(PHPLexicalGrammar.EXPRESSION_STATEMENT)
+        .is(f.expressionStatement(EXPRESSION(), EOS()));
+  }
+
+  public LabelTree LABEL() {
+    return b.<LabelTree>nonterminal(PHPLexicalGrammar.LABEL)
+        .is(f.label(b.token(PHPLexicalGrammar.IDENTIFIER), b.token(PHPPunctuator.COLON)));
+  }
+
+  public InternalSyntaxToken EOS() {
+    return b.<InternalSyntaxToken>nonterminal(PHPLexicalGrammar.EOS)
+        .is(b.firstOf(b.token(PHPPunctuator.SEMICOLON), b.token(PHPLexicalGrammar.INLINE_HTML)));
+  }
+
+  /**
+   * [ END ] Statement
+   */
+
+  public ExpressionTree EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EXPRESSION)
+      .is(f.expression(b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER)));
+  }
+
+  /**
+   * [ START ] Expression
+   */
+
+  // FIXME Fake implementation
+  public ExpressionTree UNARY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.UNARY_EXPR).is(
+      b.firstOf(
+        COMMON_SCALAR(),
+        EXPRESSION()));
+  }
+
+  public ExpressionTree CONCATENATION_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.CONCATENATION_EXPR).is(
+      f.concatenationExpr(
+        UNARY_EXPR(),
+        b.zeroOrMore(f.newTuple60(
+          b.token(DOT),
+          UNARY_EXPR()))));
+  }
+
+  public ExpressionTree MULTIPLICATIVE_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.MULTIPLICATIVE_EXPR).is(
+      f.multiplicativeExpr(
+        CONCATENATION_EXPR(),
+        b.zeroOrMore(f.newTuple61(
+          b.firstOf(b.token(STAR), b.token(DIV), b.token(MOD)),
+          CONCATENATION_EXPR()))));
+  }
+
+  public ExpressionTree ADDITIVE_EXPR() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)